pandas/tests/indexes/test_numeric.py

```diff
@@ -167,6 +167,18 @@ def test_constructor(self):
         result = Index(np.array([np.nan]))
         assert pd.isna(result.values).all()
 
+    @pytest.mark.parametrize(
+        "index, dtype",
+        [
+            (pd.Int64Index, "float64"),
+            (pd.UInt64Index, "categorical"),
+            (pd.Float64Index, "datetime64"),
+        ],
+    )
+    def test_invalid_dtype(self, index, dtype):
+        with pytest.raises(ValueError):
```

Comment:
Can you use the `match` kwarg to check the error message?  And add the GH issue as a comment?